Gems/ROS2/Code/Source/Manipulation/JointsManipulationEditorComponent.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Component/Component.h>
+#include <AzCore/std/string/string.h>
+#include <AzCore/std/containers/unordered_map.h>
+#include <AzToolsFramework/ToolsComponents/EditorComponentBase.h>
+#include <ROS2/Communication/PublisherConfiguration.h>
+
+namespace ROS2
+{
+    //! Editor Component responsible for a hierarchical system of joints such as robotic arm with Articulations or Hinge Joints.
+    class JointsManipulationEditorComponent : public AzToolsFramework::Components::EditorComponentBase
+    {
+    public:
+        JointsManipulationEditorComponent();
+        ~JointsManipulationEditorComponent() = default;
+        AZ_COMPONENT(JointsManipulationEditorComponent, "{BF2F77FD-92FB-4730-92C7-DDEE54F508BF}", AZ::Component);
```

Comment:
```suggestion
                AZ_EDITOR_COMPONENT(JointsManipulationEditorComponent, "{BF2F77FD-92FB-4730-92C7-DDEE54F508BF}");
```
Without that editor component is created in generic wraper :boom: